Core/src/IfcGeometryConverter/GeomUtils.cpp

```diff
@@ -215,49 +215,11 @@ void GeomUtils::extrude(
 	// triangulate
 	std::vector<carve::geom2d::P2> merged_path;
 	std::vector<carve::triangulate::tri_idx> triangulated;
-	std::vector<std::pair<size_t, size_t> > path_all_loops;
+	std::vector<std::pair<size_t, size_t>> path_all_loops;
 	try
 	{
 		path_all_loops = carve::triangulate::incorporateHolesIntoPolygon(face_loops);
-		// figure 2: path wich incorporates holes, described by path_all_loops
-		// (0/0) -> (1/3) -> (1/0) -> (1/1) -> (1/2) -> (1/3) -> (0/0) -> (0/1) -> (0/2) -> (0/3)
-		//  0/3<-----------------------0/2
-		//  |                            ^
-		//  |   1/0-------------->1/1    |
-		//  |   ^                   |    |
-		//  |   |                   v    |
-		//  |   1/3<--------------1/2    |
-		//  v                            |
-		//  0/0------------------------>0/1
-
-		merged_path.reserve(path_all_loops.size());
-		for( size_t i = 0; i < path_all_loops.size(); ++i )
-		{
-			int loop_number = path_all_loops[i].first;
-			int index_in_loop = path_all_loops[i].second;
-			
-			if( loop_number >= face_loops.size() )
-			{
-				err << "extrude: loop_number >= face_loops_projected.size()" << std::endl;
-				continue;
-			}
-			std::vector<carve::geom2d::P2>& loop = face_loops[loop_number];
-			carve::geom2d::P2& point_projected = loop[index_in_loop];
-			merged_path.push_back( point_projected );
-
-		}
-		// figure 3: merged path for triangulation
-		//  9<---------------------------8
-		//  |                            ^
-		//  |   2------------------>3    |
-		//  |   ^                   |    |
-		//  |   |                   v    |
-		//  |   1, 5<---------------4    |
-		//  | /                          |
-		//  0,6------------------------->7
-		carve::triangulate::triangulate(merged_path, triangulated);
-		carve::triangulate::improve(merged_path, triangulated);
-		// triangles: (9,0,1)  (5,6,7)  (4,5,7)  (4,7,8)  (9,1,2)  (8,9,2)  (3,4,8)  (2,3,8)
+		GeomUtils::createVoids(merged_path, triangulated, path_all_loops, face_loops, err);

```

Comment:
I'd name the function `incorporateVoids` since you're not creating voids, but rather just accounting for given voids.

Reply:
Resolved:  71ec6b7